strongbox-web-core/src/test/java/org/carlspring/strongbox/web/RepositoryMethodArgumentResolverTest.java

```diff
@@ -0,0 +1,331 @@
+package org.carlspring.strongbox.web;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.configuration.MutableConfiguration;
+import org.carlspring.strongbox.exception.RepositoryNotFoundException;
+import org.carlspring.strongbox.exception.ServiceUnavailableException;
+import org.carlspring.strongbox.exception.StorageNotFoundException;
+import org.carlspring.strongbox.services.ConfigurationManagementService;
+import org.carlspring.strongbox.storage.StorageDto;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.RepositoryData;
+import org.carlspring.strongbox.storage.repository.RepositoryDto;
+import org.carlspring.strongbox.storage.repository.RepositoryStatusEnum;
+
+import javax.inject.Inject;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.MethodParameter;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.web.bind.MissingPathVariableException;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.ServletWebRequest;
+import org.springframework.web.method.support.ModelAndViewContainer;
+import org.springframework.web.servlet.HandlerMapping;
+
+/**
+ * @author ankit.tomar
+ */
+@SpringBootTest
+@ContextConfiguration(classes = RepositoryMethodArgumentResolverTest.RepositoryMethodArgumentResolverConfiguration.class)
+@ActiveProfiles({ "test", "RepositoryMethodArgumentResolverTest" })
```

Comment:
we probably should have `SINGLE_THREAD` execution here.

Reply:
Fixed.

Reply:
@sbespalov : You are suggesting Single thread execution because Test case is using Mocked instances of `NativeWebRequest`, `WebDataBinderFactory`... , right?

Reply:
yes

Reply:
Thanks. :)

Reply:
you're always welcome :)